spec/features/populate_styleguide.rb

```diff
@@ -0,0 +1,217 @@
+# 100% Coverage is a good goal, but probably not necessary, nor pragmatic.
+#
+# STEP 2: DECLARATION
+# Declare all your UI Elements (name, page, selector). This file lives at /app/views/kss/home/styleguide2.html.erb
+# Example Declaration for your UI Element named "card-completed" that comes from
+# the .html file `paper_manager`
+# and it looks for css to match `source-page-selector`
+# then wraps it in a div with an ID or Class of `source-page-selector-context`
+# <div element-name="card-completed"
+#      source-page-name="paper_manager"
+#      source-page-selector=".card--completed"
+#      source-page-selector-context=".column">
+# </div>
+#
+# STEP 3: HYDRATION
+# This file is responsible for updating styleguide.html with HTML
+# from the .html files in /docs/ux, based on the declared attributes
+# This is currently `testing.rb`.
+#
+# STEP 4: UPDATING
+# This process should be run periodically
+# (every 1-2 on an active project, maybe more)
+
+
+require 'spec_helper'
+
+describe "update the Styleguide", js: true, selenium: true do
+
+  let(:admin) { FactoryGirl.create :user, :site_admin }
+  let!(:journal) { FactoryGirl.create(:journal) }
+  let!(:paper) { FactoryGirl.create(:paper, :with_tasks, journal: journal) }
+  let(:role) { FactoryGirl.create(:role, journal: journal) }
+  let!(:mmt) { FactoryGirl.create(:manuscript_manager_template, journal: journal) }
+  let!(:flow) do
+    create :flow, title: "Up for grabs", query: { assigned: true }, role_id: nil
+  end
+
+  let!(:task) { FactoryGirl.create(:financial_disclosure_task, paper: paper) }
+
+  before do
+    admin.flows << flow
+  end
+
+  # allows a block to retrieve its description - the it "is here description" do
+  let(:name) do |e|
+    e.description
+  end
+
+  before do
+    sign_in_page = SignInPage.visit
+    sign_in_page.sign_in admin
+  end
+
+  describe "save HTML and .pngs for every page in the App" do
+    scenario "dashboard" do |p|
+      visit '/'
+      page.grab(name)
+    end
+
+    scenario "navigation" do
+      visit '/'
+      find(".navigation-toggle").click
+      find(".navigation", visible: true)
+      page.grab(name, ".navigation")
+    end
+
+    scenario "flow_manager" do
+      visit '/flow_manager'
+      find(".control-bar-link", visible: true)
+      page.grab(name)
+    end
+
+    scenario "admin" do
+      visit '/admin'
+      find(".journals", visible: true)
+      page.grab(name)
+    end
+
+    # TODO
+    scenario "paper" do
+      visit '/papers/1/edit'
+      find(".manuscript-container", visible: true)
+      page.grab(name)
+    end
+
+    scenario "paper_contributors" do
+      visit '/papers/1/edit'
+      find(".manuscript-container", visible: true)
+      find(".contributors-link").click
+      find(".contributors.active")
+      page.grab(name)
+    end
+
+    scenario "paper_download_document" do
+      visit '/papers/1/edit'
+      find(".manuscript-container", visible: true)
+      find(".downloads-link").click
```

Comment:
Look into how to check for elements on screen and then interact with them more efficiently.